logic/scorer/scorer.go

```diff
@@ -15,6 +15,20 @@ func CribPoints(lead model.Card, crib []model.Card) int {
 	return points(lead, crib, true)
 }
 
+const numCardsToScore = 5
+
+// map card value (1 to 13) to number of occurrences in the hand
+// we have to include index 0 and index 14 here because some calculations pass those values in
+type valueToCount [15]uint8
+
+func (vtc valueToCount) get(value int) (uint8, bool) {
```

Comment:
:bulb: we could simplify the intent of these methods:

```go
func (vtc valueToCount) getRunMult(v int) (int, int) {
  m := vtc[v]
  if m == 0 {
    return 0, 0
  }
  r := 1
  for bv := v-1; bv > 0 && vtc[bv] > 0; bv-- {
    m *= vtc[bv]
    r++
  }
  for av := v+1; av > 0 && vtc[av] > 0; av++ {
    m *= vtc[av]
    r++
  }
  return r, m
}
```

with

```go
func scoreRuns(values [numCardsToScore]int, valuesToCounts valueToCount) (scoreType, int) {
	for _, v := range values {
		r, m := valuesToCounts.getRunMult(v)
		if r >= 3 {
			return calculateTypeAndPoints(r, m)
		}
	}
	return 0, 0
}
```

Reply:
Hmmm I'm not sure how much simpler `getRunMult` is compared to what's currently in `scoreRuns` . The logic to get run and multiplier aren't inherently related to the `valueToCount` type in any way, so I'm reluctant to make them methods on the type. Why are you looking to get rid of `get`?

Reply:
FWIW I got rid of the type and methods and just did things manually in `scoreRuns`